Makefile

```diff
@@ -11,36 +11,29 @@ get-deps:
 	$(REBAR) get-deps
 
 compile:
-	$(REBAR) compile
-	$(REBAR) skip_deps=true xref
+	$(REBAR) compile xref
 
 quick:
-	$(REBAR) skip_deps=true compile
-	$(REBAR) skip_deps=true xref
+	$(REBAR) compile xref
 
 clean:
 	$(REBAR) clean
-	rm -f erl_cache
+	rm -fr _build
 
 test: compile
-	$(REBAR) skip_deps=true eunit
+	$(REBAR) eunit --cover
 
 docs: docsclean
-	ln -s . doc/doc
-	$(REBAR) skip_deps=true doc
+	$(REBAR) edoc
 
 docsclean:
 	rm -f doc/*.html doc/*.css doc/erlang.png doc/edoc-info doc/doc
 
 go:
-	erl -name erl_cache -pa deps/*/ebin -pa ebin/ -s erl_cache start ${EXTRA_ARGS}
+	$(REBAR) shell
 
-dialyzer:
-	dialyzer -c ebin/ -Wunmatched_returns -Werror_handling -Wrace_conditions
+.erl_cache.plt:
+	dialyzer --output_plt $@ --build_plt --apps erts kernel stdlib
 
-erl_cache:
-	REBAR_BENCH=1 $(REBAR) get-deps compile
-	REBAR_BENCH=1 $(REBAR) escriptize skip_deps=true
-
-benchmark: erl_cache quick
-	./erl_cache priv/bench.conf
+dialyzer: .erl_cache.plt quick
```

Comment:
nit: I suggest removing `quick` target and use `compile` as a dependency to this target, since `quick` is quite confusing: I initially assumed `quick` is a "lightweight linter", like xref.

Reply:
@motiejus Sure, I don't really mind. `quick` is pretty much a leftover from rebar2 times, and I didn't want to remove it in case the `make quick` command was linked to any organizational process.
I'll gladly remove `quick` and use `compile` as a `dialyzer` dependency